store/store.go

```diff
@@ -511,3 +514,16 @@ func indexTo[H header.Header[H]](ctx context.Context, batch datastore.Batch, hea
 	}
 	return nil
 }
+
+// sleep with cancellation, returns nil when timer has fired, error otherwise.
+func sleep(ctx context.Context, duration time.Duration) error {
```

Comment:
Sadly this func isn't presented in stdlib `sync` package :( 

CC: https://github.com/cristalhq/synx